packages/glimmer-runtime/lib/environment.ts

```diff
@@ -258,20 +251,21 @@ export abstract class Environment {
     this.scheduledUpdateModifiers = null;
   }
 
-  abstract hasHelper(helperName: string[], symbolTable: SymbolTable): boolean;
-  abstract lookupHelper(helperName: string[], symbolTable: SymbolTable): Helper;
-
   attributeFor(element: Simple.Element, attr: string, isTrusting: boolean, namespace?: string): IChangeList {
     return defaultChangeLists(element, attr, isTrusting, namespace);
   }
 
-  abstract hasPartial(partialName: string[], symbolTable: SymbolTable): boolean;
-  abstract lookupPartial(PartialName: string[], symbolTable: SymbolTable): PartialDefinition<TemplateMeta>;
+  abstract hasHelper(helperName: string[], blockMeta: TemplateMeta): boolean;
+  abstract lookupHelper(helperName: string[], blockMeta: TemplateMeta): Helper;
+
+  abstract hasModifier(modifierName: string[], blockMeta: TemplateMeta): boolean;
+  abstract lookupModifier(modifierName: string[], blockMeta: TemplateMeta): ModifierManager<Opaque>;
+
   abstract hasComponentDefinition(tagName: string[], symbolTable: SymbolTable): boolean;
   abstract getComponentDefinition(tagName: string[], symbolTable: SymbolTable): ComponentDefinition<Opaque>;
```

Comment:
I want to change the rest of these (helpers, modifiers, components) from `string[]` to `string`. The only issue is we need to make `{{f.foo with some="args"}}` illegal in the Glimmer compiler, but Ember actually use that feature. The plan is the make those errors happen in the "JavaScript" compiler, so Ember would have a chance to transform it into `{{component f.foo some="args"}}` before it hits the error. Or we could just allow it for helpers only, but the inconsistency bugs me a little.